builtin/mainmenu/dlg_mg_flags.lua

```diff
@@ -0,0 +1,211 @@
+--it under the terms of the GNU Lesser General Public License as published by
+--the Free Software Foundation; either version 2.1 of the License, or
+--(at your option) any later version.
+--
+--This program is distributed in the hope that it will be useful,
+--but WITHOUT ANY WARRANTY; without even the implied warranty of
+--MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--GNU Lesser General Public License for more details.
+--
+--You should have received a copy of the GNU Lesser General Public License along
+--with this program; if not, write to the Free Software Foundation, Inc.,
+--51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+local mg_first_time = true
+
+-- use default if not defined
+
+-- water level
+local water_level = 1
+
+-- general attributes
+local flag_list_g = {
+	trees		= true,
+	caves		= true,
+	flat		= false,
+	dungeons	= false,
+	light		= true
+}
+
+-- v6 mapgen
+local flag_list_6 = {
+	biomeblend	= true,
+	jungles		= true,
+	snowbiomes	= true,
+	mudflow		= true
+}
+
+-- v7 mapgen
+local flag_list_7 = {
+	mountains	= true,
+	ridges		= true
+}
+
+local function update_mg_flags()
+	local function mg_flags_helper(flag_list)
+		local first = true
+		local flag_str = ""
+		for flag, val in pairs(flag_list) do
+			flag_str = flag_str .. (first and "" or ", ") .. (val == true and "" or "no") .. flag
+			first = false
+		end
+		return flag_str
+	end
+	
+	core.setting_set("mg_flags", mg_flags_helper(flag_list_g))
+	core.setting_set("mgv6_spflags", mg_flags_helper(flag_list_6))
+	core.setting_set("mgv7_spflags", mg_flags_helper(flag_list_7))
+	
+	core.setting_set("water_level", water_level)
+end
+
+-- load default settings if not exist
+if core.setting_get("water_level") == nil then
+	core.setting_set("water_level",1)
+end
+if core.setting_get("mg_flags") == nil then
+	core.setting_set("mg_flags","trees, caves")
+end
+if core.setting_get("mgv6_spflags") == nil then
+	core.setting_set("mgv6_spflags","biomeblend, jungles, mudflow")
+end
+if core.setting_get("mgv7_spflags") == nil then
+	core.setting_set("mgv7_spflags","mountains, ridges")
+end
+
+local function load_mg_flags()
+	local function mg_flags_helper(flag_list,text)
+		local text = string.split(text, ", ")
+		for i = 1, #text do
+			if text[i]:sub(1, 2) == "no" then
+				flag_list[text[i]:sub(3)] = false
+			else
+				flag_list[text[i]] = true
+			end
+		end
+	end
+	mg_flags_helper(flag_list_g, core.setting_get("mg_flags"))
+	mg_flags_helper(flag_list_6, core.setting_get("mgv6_spflags"))
+	mg_flags_helper(flag_list_7, core.setting_get("mgv7_spflags"))
+	
+	water_level = core.setting_get("water_level")
+	
+	update_mg_flags()
+end
+
+local function mg_flags_formspec(dialogdata)
+
+	-- load minetest.conf for first time
+	if mg_first_time == true then
+		load_mg_flags()
+		update_mg_flags()
+	end
+	mg_first_time = false
+	
+	-- this is it!
+	local formspec =
+		"size[8,4,true]" ..
+		"button[4,3.65;3,0.5;mg_ok;" .. fgettext("Ok") .. "]" ..
```

Comment:
I think "Close" is a better name for this, as the changes take effect already after the person has made the change.